src/netlink.c

```diff
@@ -503,18 +521,26 @@ static int qos_filter_cb(const struct nlmsghdr *nlh, void *args) {
       return MNL_CB_ERROR;
     }
 
-    mnl_attr_parse_nested(attr, qos_attr_cb, &bs);
+    mnl_attr_parse_nested(attr, qos_attr_cb, &stats);
 
-    if (bs != NULL) {
+    if (stats.bstats != NULL || stats.qstats != NULL) {
       char type_instance[DATA_MAX_NAME_LEN];
 
       stats_submitted = 1;
 
       ssnprintf(type_instance, sizeof(type_instance), "%s-%s", tc_type,
                 tc_inst);
-
-      submit_one(dev, "ipt_bytes", type_instance, bs->bytes);
-      submit_one(dev, "ipt_packets", type_instance, bs->packets);
+      if (stats.bstats != NULL) {
+        submit_one(dev, "ipt_bytes", type_instance, stats.bstats->bytes);
+        submit_one(dev, "ipt_packets", type_instance, stats.bstats->packets);
+      }
+      if (stats.qstats != NULL) {
+        submit_one(dev, "ipt_qlen", type_instance, stats.qstats->qlen);
+        submit_one(dev, "ipt_backlog", type_instance, stats.qstats->backlog);
```

Comment:
@octo As far as I understand, the significance of qlen and backlog differs between queueing algorithms. _qlen_ is not synonymous with maximum buffer length, in some types of qdiscs, it is the length of the standing queue (packets), it is a gauge, not a delta. Reading the source of the queue implementation is required to fully understand its details.

We use primarily HFSC, and as far as I can tell, _qlen_ seems to be used as an internal metric only, while _backlog_ seems to be a value that is exported to statistics only. (even though both seem to be readable through netlink)